### PR TITLE
Make nodeagent debug endpoint dump cert data in base64

### PR DESCRIPTION
**Pull request by Monkeyanator**
For comparison with secret configuration dump from Envoy, it makes more sense
to have the debug endpoint dump the base64 encoded x509 certs. In
addition, add a guard to ensure no null deref in DebugInfo func.

/cc @JimmyCYJ 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: security/pkg/nodeagent/sds/sdsservice.go [hunk 1]
```diff
@@ -160,16 +161,21 @@ func (s *sdsservice) register(rpcs *grpc.Server) {
...
-			ConnectionID:     connKey.ConnectionID,
-			ProxyID:          conn.proxyID,
-			ResourceName:     conn.ResourceName,
-			CertificateChain: string(conn.secret.CertificateChain),
-			RootCert:         string(conn.secret.RootCert),
+			ConnectionID: connKey.ConnectionID,
+			ProxyID:      conn.proxyID,
+			ResourceName: conn.ResourceName,
+			// should be base64 encoded for straightforward comparison with Envoy SDS config dump format
+			CertificateChain: base64.StdEncoding.EncodeToString(conn.secret.CertificateChain),
```
Comment (incfly) [inline security/pkg/nodeagent/sds/sdsservice.go]: What's the wire format we send in the SDS response payload?

I'm wondering where the discrepancy is coming from. Could be

1. Envoy config dump library already pre-processed all the string, making raw bytes as base64 decoded.
1. Somewhere in the node agent code, we encode base64 before setting the response.

If it's 2, might worth seeing can we just use one base64 encoder in one place.

Forget about this if it's too involved.

Comment (Monkeyanator) [reply security/pkg/nodeagent/sds/sdsservice.go]: From what I can tell ([from following the code path here](https://github.com/istio/istio/blob/3c588f9acde157b0e43f99ac9b97d127c55f43fb/security/pkg/nodeagent/sds/sdsservice.go#L310)), the SDS response gives raw, non-base64 encoded secret data back to Envoy, so it seems like the encoding must be happening on Envoy-side before the config dump.

**Comment by incfly** [comment]
/lgtm
/approve

All right, then don't worry about it.

**Pull request merged**